src/vs/workbench/contrib/terminal/node/terminalProfiles.ts

```diff
@@ -0,0 +1,198 @@
+}
+
+async function detectAvailableWindowsProfiles(detectWslProfiles?: boolean, statProvider?: IStatProvider): Promise<ITerminalProfile[]> {
+	// Determine the correct System32 path. We want to point to Sysnative
+	// when the 32-bit version of VS Code is running on a 64-bit machine.
+	// The reason for this is because PowerShell's important PSReadline
+	// module doesn't work if this is not the case. See #27915.
+	const is32ProcessOn64Windows = process.env.hasOwnProperty('PROCESSOR_ARCHITEW6432');
+	const system32Path = `${process.env['windir']}\\${is32ProcessOn64Windows ? 'Sysnative' : 'System32'}`;
+
+	let useWSLexe = false;
+
+	if (getWindowsBuildNumber() >= 16299) {
+		useWSLexe = true;
+	}
+
+	let expectedProfiles: IPotentialTerminalProfile[] = [
+		{
+			profileName: 'Command Prompt',
+			paths: [`${system32Path}\\cmd.exe`]
+		},
+		{
+			profileName: 'Git Bash',
+			paths: [
+				`${process.env['ProgramW6432']}\\Git\\bin\\bash.exe`,
+				`${process.env['ProgramW6432']}\\Git\\usr\\bin\\bash.exe`,
+				`${process.env['ProgramFiles']}\\Git\\bin\\bash.exe`,
+				`${process.env['ProgramFiles']}\\Git\\usr\\bin\\bash.exe`,
+				`${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe`,
+			],
+			args: ['--login']
+		},
+		{
+			profileName: 'Cygwin',
+			paths: [
+				`${process.env['HOMEDRIVE']}\\cygwin64\\bin\\bash.exe`,
+				`${process.env['HOMEDRIVE']}\\cygwin\\bin\\bash.exe`
+			],
+			args: ['-l']
+		},
+		... await getWslProfiles(system32Path, useWSLexe, detectWslProfiles),
+		... await getPowershellProfiles()
+	];
+
+	const promises: Promise<ITerminalProfile | undefined>[] = [];
+	expectedProfiles.forEach(profile => promises.push(validateProfilePaths(profile.profileName, profile.paths, statProvider, profile.args)));
+	const profiles = await Promise.all(promises);
+
+	// todo what's the expected path of zsh? add to expected profiles
+	let zsh = profiles.find(shell => shell?.path.endsWith('zsh.exe'));
+	if (zsh) {
+		zsh.args = ['--login'];
+	}
+	return coalesce(profiles);
+}
+
+async function getPowershellProfiles(): Promise<IPotentialTerminalProfile[]> {
+	let profiles: IPotentialTerminalProfile[] = [];
+	// Add all of the different kinds of PowerShells
+	for await (const pwshExe of enumeratePowerShellInstallations()) {
+		profiles.push({ profileName: pwshExe.displayName, paths: [pwshExe.exePath] });
+	}
+	return profiles;
+}
+
+async function getWslProfiles(path: string, useWSLexe: boolean, detectWslProfiles?: boolean): Promise<IPotentialTerminalProfile[]> {
+	if (detectWslProfiles) {
+		let profiles: IPotentialTerminalProfile[] = [];
+		profiles.push({ profileName: `WSL Bash`, paths: [`${path}\\${useWSLexe ? 'wsl.exe' : 'bash.exe'}`] });
+		const distroOutput = await new Promise<string>(r => cp.exec('wsl.exe -l', (err, stdout) => err ? console.trace('problem occurred when getting wsl distros', err) : r(stdout)));
+		if (distroOutput) {
+			let regex = new RegExp(/[\r\n]/);
```

Comment:
I think this is right?

```suggestion
			let regex = new RegExp(/\r?\n/);
```